src/main/java/Deadline.java

```diff
@@ -0,0 +1,118 @@
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Represents a task with a deadline.
+ */
+public class Deadline extends Task {
+    protected String by;
+    protected LocalDate byDate;
+    protected LocalDateTime byDateTime;
+
+    /**
+     * Initializes a deadline task containing the task description and deadline of the task.
+     *
+     * @param description Description of the deadline task.
+     * @param by          Deadline of the task.
+     */
+    public Deadline(String description, String by) {
+        super(description);
+
+        this.byDateTime = tryParseDateTime(by);
+        if (byDateTime == null) {
+            this.byDate = tryParseDate(by);
+        }
+        this.by = by;
+    }
+
+    /**
+     * Initializes a deadline task containing the task description, if the task is done and deadline of the task.
+     * This is an overloaded constructor to allow for tasks in the hard drive to be loaded when Duke first runs.
+     *
+     * @param description Description of the deadline task.
+     * @param isDone      If task is done.
+     * @param by          Deadline of the task.
+     */
+    public Deadline(String description, boolean isDone, String by) {
+        super(description, isDone);
+        this.byDateTime = tryParseDateTime(by);
+        if (byDateTime == null) {
+            this.byDate = tryParseDate(by);
+        }
+        this.by = by;
+    }
+
+    /**
+     * Parses the date and time string, and returns a LocalDateTime object if the date and time string is in accordance
+     * to any of the listed formats.
+     * If date and time string does not follow any of the listed formats, null is returned.
+     *
+     * @param dateString String representing the date and time.
+     * @return LocalDateTime object if parsing the successful.
+     */
+    public LocalDateTime tryParseDateTime(String dateString) {
+        List<String> formatStrings = Arrays.asList("yyyy-MM-dd HHmm", "yyyy-MM-d HHmm", "dd/MM/yyyy HHmm",
```

Comment:
Great effort here to format all the different types of DateTime to parse for the user. It is user friendly :)